src/collections/maple_tree/mod.rs

```diff
@@ -273,5 +742,285 @@ mod tests {
+        // and returns the first value that is found outside the range.
+
+        sl.insert(10, 10);
+        sl.insert(11, 11);
+        sl.insert(12, 12);
+        assert!(sl.check_bounds(&0, &8) == false);
+    }
+
+    #[test]
+    fn test_sparse_leaf_sort() {
+        // test sorting full node
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
+        let mut test_vals: [usize; 8] = [3, 8, 7, 4, 2, 1, 5, 6];
+        let mut sorted_test_vals: [usize; 8] = [1, 2, 3, 4, 5, 6, 7, 8];
+
+        for val in test_vals.iter() {
+            sl.insert(*val, *val);
+        }
+        for i in 0..CAPACITY {
+            match sl.key[i] {
+                M::Some(v) => {
+                    println!("{0}", v);
+                }
+                M::None => println!("None"),
+            }
+        }
+        sl.sort();
+
+        for i in 0..CAPACITY {
+            // the code inside match is usefull for debuging if a test fails
+            match sl.key[i] {
+                M::Some(v) => {
+                    println!(
+                        "(actualValue = {0}) - (sortedTestValue = {1})",
+                        v, sorted_test_vals[i]
+                    );
+                }
+                M::None => println!("None - {}", sorted_test_vals[i]),
+            }
+            assert!(sl.key[i] == M::Some(sorted_test_vals[i]));
+        }
+
+        // test sorting half full node with M::None's inbetween each value
+        // i.e [3, None, 4, None, 2, None, 1, None]
+
+        test_vals = [3, 8, 4, 6, 2, 7, 1, 5];
+        let none_positions: [usize; 4] = [8, 6, 7, 5];
+        sl = SparseLeaf::new();
+
+        for val in test_vals.iter() {
+            sl.insert(*val, *val);
+        }
+
+        // remove every second value from sl
+        for val in none_positions.iter() {
+            sl.remove(&val);
+        }
+
+        sl.sort();
+
+        for i in 0..4 {
+            println!("{} <-> ", sorted_test_vals[i]);
+            assert!(sl.key[i] == M::Some(sorted_test_vals[i]));
+        }
+    }
+
+    #[test]
+    fn test_range_node_find_range(){
+        
+        let mut rn: RangeLeaf<usize, usize> = RangeLeaf::new();
+        let min: usize = 3;
+        let max: usize = 100;
+ 
+        rn.pivot[0] = M::Some(7);
+        rn.pivot[1] = M::Some(11);
+        rn.pivot[2] = M::Some(20);
+        rn.pivot[3] = M::Some(25);
+        rn.pivot[4] = M::Some(35);
+        rn.pivot[5] = M::Some(69);
+        rn.pivot[6] = M::Some(85);
+        
+
+        assert!(rn.find_range(&3, &4, &min, &max) == Min(0));
+        assert!(rn.find_range(&5, &17, &min, &max) == Min(2));
+        assert!(rn.find_range(&7, &25, &min, &max) == Pivots(0, 3));
+        assert!(rn.find_range(&3, &100, &min, &max) == MinMax); 
+        assert!(rn.find_range(&90, &95, &min, &max) == Max(6));
+        assert!(rn.find_range(&20, &95, &min, &max) == Max(2));
+
+    }
+
+    #[test]
+    fn test_range_node_move_pivots_left(){
```

Comment:
You may consider adding test for:

* Get single value
* Insert single value
* delete single value
* iterate over range(start -> end).

Reply:
I am about to start writing the code for insert so I will add the tests for that as I go.

Reply:
Sounds good!